frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -589,5 +589,6 @@ export type SelfSubjectAccessReviewKind = {
 export type ResourceYAMLEditorProps = {
   initialResource: string | { [key: string]: any };
   header?: string;
+  error?: string;
   onSave?: (content: string) => void;
 };
```

Comment:
```suggestion
export type ResourceYAMLEditorProps = {
  initialResource: string | { [key: string]: any };
  header?: string;
  onSave?: (content: string) => void;
} | {
  initialResource: string | { [key: string]: any };
  header?: string;
  error?: string;
  onSave: (content: string) => void;
};
```
This would make sure that when user pass `error` they pass an `onSave` too.

Reply:
@spadgett thoughts?